runtime/gc_base/ObjectAccessBarrier.cpp

```diff
@@ -1552,10 +1584,47 @@ MM_ObjectAccessBarrier::cloneIndexableObject(J9VMThread *vmThread, J9IndexableOb
 		_extensions->indexableObjectModel.memcpyArray(destObject, srcObject);
 	}
 
-	/* zero lockword, if present */
+	/* initialize lockword, if present */
+	J9Class *objectClass = J9GC_J9OBJECT_CLAZZ_THREAD(destObject, vmThread);
 	lockwordAddress = getLockwordAddress(vmThread, (J9Object*) destObject);
 	if (NULL != lockwordAddress) {
-		J9_STORE_LOCKWORD(vmThread, lockwordAddress, OBJECT_HEADER_LOCK_RESERVED);
+		U_32 lwValue = 0;
```

Comment:
Is this code copy/paste from copyObjectFields()? If it is identical please put it in the function

Reply:
Yes, the code for determining the initial lockword is very similar and occurs in a few places. I will try to combine them into one place.